integration/update_field_test.go

```diff
@@ -0,0 +1,929 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"math"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/FerretDB/FerretDB/integration/shareddata"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/testutil"
+)
+
+// This file is for field update operator tests:
+// - `$currentDate`
+// - `$inc`
+// - `$min`
+// - `$max`
+// - `$mul`
+// - `$rename`
+// - `$set`
+// - `$setOnInsert`
+// - `$unset`.
+
+func TestUpdateFieldCurrentDateTimestamp(t *testing.T) {
```

Comment:
The test name should be `TestUpdateField<OperatorName>`. All variations of it (TestUpdateFieldCurrentDateTimestamp and TestUpdateFieldCurrentDate, TestUpdateFieldIncErrors and TestUpdateFieldInc below, etc) should be subtests

Reply:
> The test name should be `TestUpdateField<OperatorName>`. All variations of it (TestUpdateFieldCurrentDateTimestamp and TestUpdateFieldCurrentDate, TestUpdateFieldIncErrors and TestUpdateFieldInc below, etc) should be subtests

done.

Reply:
@AlekSi should it be done for all other update tests too? Did I get it right that the expected result is that we have one test per iterator?

Reply:
I think we should have one top-level test per specific operator. There can be a separate test for a mix of operators.

---

@seeforschauer 

> TestUpdateFieldCurrentDateTimestamp and TestUpdateFieldCurrentDate

Those tests were merged, great

> TestUpdateFieldIncErrors and TestUpdateFieldInc below

Those two were not merged, so that's not done

We also have tests that do not use shareddata.Scalars.

Reply:
> We also have tests that do not use shareddata.Scalars. 

Could you please clarify, 
`shareddata.Scalars` must be removed from tests that do not require them?